src/ORM/DataList.php

```diff
@@ -1055,7 +1055,14 @@ private function fetchEagerLoadRelations(Query $query): void
             $parentIDs = $topLevelIDs;
             $parentRelationData = $query;
             $chainToDate = [];
+            $polymorphicEncountered = false;
             foreach (explode('.', $relationChain) as $relationName) {
+                if ($polymorphicEncountered) {
+                    $polymorphicRelation = $chainToDate[count($chainToDate) - 1];
```

Comment:
```suggestion
                    $polymorphicRelation = $chainToDate[array_key_last($chainToDate)];
```
This is a stylistic suggestion so I won't block merging if it's not done, but it is more easily readable at a glance.